### PR TITLE
Adding logic to capture if a datamodel is a string and parse it as JS…

### PR DESCRIPTION
**Pull request by ivanwills**
…ON (if any errors the datamodel is left untouched

### CODE REVIEW

**Comment by scrypto** [comment]
👍🏻

File: src/ractivef.base.js [hunk 1]
```diff
@@ -22,10 +22,9 @@ Ractive.defaults.onconstruct = function (opts) {
 	if (opts.data && opts.data.datamodel) {
 		if (typeof opts.data.datamodel === 'string') {
 			try {
-				var newModel = opts.data.datamodel.replace(/&#(\d+);/g, function (match, dec) {
-					return String.fromCharCode(dec);
-				}).replace(/&quot;/g, '"');
-				opts.data.datamodel = JSON.parse(newModel);
+				var newModel = document.createElement('textarea');
+				newModel.innerHTML = opts.data.datamodel;
+				opts.data.datamodel = JSON.parse(newModel.value);
```
Comment (scott-dickson) [inline src/ractivef.base.js:27]: Just thinking do you want to then remove the textarea now that it's been finished with?

**Review by scott-dickson — COMMENTED** [review]
Just thinking do you want to then remove the textarea now that it's been finished with?

File: src/ractivef.base.js (hunk 1 above)
Comment (scott-dickson) [reply src/ractivef.base.js:27]: Doh... it's a local variable! My bad

**Review by scott-dickson — APPROVED** [review]
Looks good.  Thanks Ivan.

**Review by SylvieTeoOptus — APPROVED** [review]

**Review by shuyiwu — APPROVED** [review]

**Pull request merged**